module/actor/witcherActor.js

```diff
@@ -25,7 +24,266 @@ export default class WitcherActor extends Actor {
+    let curSpd = Math.floor((this.system.stats.spd.max + spdTotalModifiers - encDiff) / spdDivider);
+    let curEmp = Math.floor((this.system.stats.emp.max + empTotalModifiers) / empDivider);
+    let curCra = Math.floor((this.system.stats.cra.max + craTotalModifiers) / craDivider);
+    let curWill = Math.floor((this.system.stats.will.max + willTotalModifiers) / willDivider);
+    let curLuck = Math.floor((this.system.stats.luck.max + luckTotalModifiers) / luckDivider);
+    let isDead = false;
+    let isWounded = false;
+    let HPvalue = this.system.derivedStats.hp.value;
+    if (HPvalue <= 0) {
+      isDead = true
+      curInt = Math.floor((this.system.stats.int.max + intTotalModifiers) / 3 / intDivider)
+      curRef = Math.floor((this.system.stats.ref.max + refTotalModifiers - armorEnc - encDiff) / 3 / dexDivider)
+      curDex = Math.floor((this.system.stats.dex.max + dexTotalModifiers - armorEnc - encDiff) / 3 / refDivider)
+      curBody = Math.floor((this.system.stats.body.max + bodyTotalModifiers) / 3 / bodyDivider)
+      curSpd = Math.floor((this.system.stats.spd.max + spdTotalModifiers - encDiff) / 3 / spdDivider)
+      curEmp = Math.floor((this.system.stats.emp.max + empTotalModifiers) / 3 / empDivider)
+      curCra = Math.floor((this.system.stats.cra.max + craTotalModifiers) / 3 / craDivider)
+      curWill = Math.floor((this.system.stats.will.max + willTotalModifiers) / 3 / willDivider)
+      curLuck = Math.floor((this.system.stats.luck.max + luckTotalModifiers) / 3 / luckDivider)
+    }
+    else if (HPvalue < this.system.coreStats.woundTreshold.current > 0) {
+      isWounded = true
+      curRef = Math.floor((this.system.stats.ref.max + refTotalModifiers - armorEnc - encDiff) / 2 / refDivider)
+      curDex = Math.floor((this.system.stats.dex.max + dexTotalModifiers - armorEnc - encDiff) / 2 / dexDivider)
+      curInt = Math.floor((this.system.stats.int.max + intTotalModifiers) / 2 / intDivider)
+      curWill = Math.floor((this.system.stats.will.max + willTotalModifiers) / 2 / willDivider)
+    }
+
+    let hpTotalModifiers = this.getAllModifier("hp").totalModifiers;
+    let staTotalModifiers = this.getAllModifier("sta").totalModifiers;
+    let resTotalModifiers = this.getAllModifier("resolve").totalModifiers;
+    let focusTotalModifiers = this.getAllModifier("focus").totalModifiers;
+    let vigorModifiers = this.getAllModifier("vigor").totalModifiers;
+    let hpDivider = this.getAllModifier("hp").totalDivider;
+    let staDivider = this.getAllModifier("sta").totalDivider;
+    this.system.derivedStats.hp.modifiers.forEach(item => hpTotalModifiers += Number(item.value));
+    this.system.derivedStats.sta.modifiers.forEach(item => staTotalModifiers += Number(item.value));
+    this.system.derivedStats.resolve.modifiers.forEach(item => resTotalModifiers += Number(item.value));
+    this.system.derivedStats.focus.modifiers.forEach(item => focusTotalModifiers += Number(item.value));
```

Comment:
Here you can use it as well. Since there are already several places, where it can be used, you can make it a generic function:

```javascript
function getModifiersMap(statsObj) {
  Object.keys(statsObj)
    .filter(stat => Array.isArray(stat.modifiers)) // Only take stats that have modifiers
    .map(stat => {
      const totalValue = stat.modifiers.reduce((total, mod) => (total += mod.value), this.getAllModifier(stat)?.totalModifiers || 0);
      return [stat, totalValue];
    });
}
```

Reply:
https://github.com/witchertrpg-foundryvtt/TheWitcherTRPG/issues/61